central/resourcecollection/datastore/datastore_impl.go

```diff
@@ -3,18 +3,120 @@ package datastore
 import (
 	"context"
 
+	"github.com/heimdalr/dag"
 	"github.com/stackrox/rox/central/resourcecollection/datastore/index"
 	"github.com/stackrox/rox/central/resourcecollection/datastore/search"
-	"github.com/stackrox/rox/central/resourcecollection/datastore/store/postgres"
+	"github.com/stackrox/rox/central/resourcecollection/datastore/store"
 	v1 "github.com/stackrox/rox/generated/api/v1"
 	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/logging"
 	pkgSearch "github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+var (
+	log           = logging.LoggerForModule()
+	initBatchSize = 20
 )
 
 type datastoreImpl struct {
-	storage  postgres.Store
+	storage  store.Store
 	indexer  index.Indexer
 	searcher search.Searcher
+
+	graphLock sync.Mutex
+	graph     *dag.DAG
+}
+
+func resetLocalGraph(ds *datastoreImpl) {
```

Comment:
If this is only used for tests, you can move it to the test file since it's in the same package. Reduces risk of it accidentally getting used in the main file